dash/development/_r_components_generation.py

```diff
@@ -0,0 +1,531 @@
+        if 'defaultValue' in props[p] else
+        '{}=NULL'.format(p)
+        for p in prop_keys
+    )
+
+    item_text += "\n\n".join(
+        '\\item{{{}}}{{{}}}'.format(p, props[p]['description'])
+        for p in prop_keys
+    )
+
+    file_path = os.path.join('man', file_name)
+    with open(file_path, 'w') as f:
+        f.write(help_string.format(
+            prefix=prefix,
+            name=name,
+            default_argtext=default_argtext,
+            item_text=item_text,
+            description=description.replace('\n', ' ')
+        ))
+
+
+def write_class_file(name,
+                     props,
+                     description,
+                     project_shortname,
+                     prefix=None):
+    import_string =\
+        "# AUTO GENERATED FILE - DO NOT EDIT\n\n"
+    class_string = generate_class_string(
+        name,
+        props,
+        project_shortname,
+        prefix
+    )
+    file_name = "{}{}.R".format(prefix, name)
+
+    file_path = os.path.join('R', file_name)
+    with open(file_path, 'w') as f:
+        f.write(import_string)
+        f.write(class_string)
+
+    # generate the R help pages for each of the Dash components that we
+    # are transpiling -- this is done to avoid using Roxygen2 syntax,
+    # we may eventually be able to generate similar documentation using
+    # doxygen and an R plugin, but for now we'll just do it on our own
+    # from within Python
+    write_help_file(
+        name,
+        props,
+        description,
+        prefix
+    )
+
+    print('Generated {}'.format(file_name))
+
+
+# pylint: disable=inconsistent-return-statements
+def generate_export_string(name, prefix):
+    if not name.endswith('-*') and \
+            str(name) not in r_keywords and \
+            str(name) not in ['setProps', 'children', 'dashEvents']:
+        return 'export({}{})\n'.format(prefix, name)
+
+
+def write_js_metadata(project_shortname):
+    """
+    Write an internal (not exported) R function to return all JS
+    dependencies as required by htmltools package given a
+    function string
+
+    Parameters
+    ----------
+    project_shortname = hyphenated string, e.g. dash-html-components
+
+    Returns
+    -------
+
+    """
+    function_string = generate_js_metadata(
+        project_shortname
+    )
+    file_name = "internal.R"
+
+    # the R source directory for the package won't exist on first call
+    # create the R directory if it is missing
+    if not os.path.exists('R'):
+        os.makedirs('R')
+
+    file_path = os.path.join('R', file_name)
+    with open(file_path, 'w') as f:
+        f.write(function_string)
+
+    # now copy over all JS dependencies from the (Python) components dir
+    # the inst/lib directory for the package won't exist on first call
+    # create this directory if it is missing
+    if not os.path.exists('inst'):
+        os.makedirs('inst')
+
+    if not os.path.exists('inst/lib'):
+        os.makedirs('inst/lib')
```

Comment:
You can just do os.makedirs('inst/lib') no need for the first call, makedirs will create the whole folder structure.